src/components/calendar/calendar.jsx

```diff
@@ -76,6 +76,24 @@ function Calendar(props) {
   const [activeDate, setActiveDate] = useState(defaultFocusedDate);
   let refs = {};
 
+  useEffect(() => {
+    if (selectedDate && defaultDate && !isSameDate(selectedDate, defaultDate)) {
+      setSelectedDate(defaultDate);
+      setFocusedDate(defaultDate);
+
+      setYear(highlightedDate.getFullYear());
+      setMonth(highlightedDate.getMonth());
+      setActiveDate(defaultDate);
+    }
+  }, [defaultDate]);
```

Comment:
We might want to only run this when we get a new `props.value` instead of a new `new Date` object. Every date object is different.

e.g. 

```
> new Date() === new Date()
< false
```

Reply:
That, or if we want to maintain running it every time the `render` is invoked, we could just move it to an IIFE.